Fix module resolution error in package

## Problem
There's an issue with the web-link-collector package (v1.0.1) related to module resolution:

1. The package is configured as an ESM package (`"type": "module"` in package.json)
2. The main entry point is set to `"main": "dist/index.js"`, but this file doesn't exist in the package
3. When importing the package, Node.js throws an error:
   ```
   Error: Cannot find module '/path/to/node_modules/web-link-collector/dist/index.js'
   ```

## Root Cause
The TypeScript compiler outputs the compiled JavaScript files to `dist/src/index.js` but the package.json incorrectly points to a non-existent file.

## Steps to Reproduce

1. Install the package: `npm install web-link-collector`
2. Create a TypeScript or JavaScript file that imports from the package:
   ```typescript
   import { collectLinks } from 'web-link-collector';
   ```
3. Run the file with Node.js

## Expected Behavior
The module should be imported successfully without any errors.

## Actual Behavior
The import fails with an error about missing module.

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "web-link-collector",
   "version": "1.0.1",
   "description": "A library and CLI tool to recursively collect links from a given initial URL and output them as structured data",
-  "main": "dist/index.js",
+  "main": "dist/src/index.js",
   "types": "dist/src/index.d.ts",
   "type": "module",
   "bin": {

